registry/config.go

```diff
@@ -150,6 +151,19 @@ skip:
 			config.ServiceConfig.IndexConfigs = originalIndexInfos
 			return err
 		}
+		if strings.HasPrefix(r, "http://") {
```

Comment:
Are schemes case-sensitive? If not we could consider using a similar approach as [`ValidateMirror()`](https://github.com/docker/docker/blob/d3f30d6a67577a52162b36a1af13eb4167e48c6b/registry/config.go#L239), although i'm not fond of that function doing both validation and normalization.

I'm okay with doing that in a follow up / separate as it's really a corner case